cmd/stats/stats.go

```diff
@@ -2,28 +2,223 @@ package stats
+
+		if err != nil {
+			return todayAtMidnight, fmt.Errorf("error parsing date: %s", err)
+		}
+
+		argTime = time.Date(
+			argTime.Year(), argTime.Month(), argTime.Day(),
+			0, 0, 0, 0, now.Location())
+
+		if !isStart {
+			argTime = argTime.AddDate(0, 0, 1).Add(-1 * time.Nanosecond)
+		}
+
+		if argTime.After(now) {
+			return todayAtMidnight, fmt.Errorf("invalid date: %s hasn't happend yet!", arg)
+		}
+
+		return argTime, nil
+	}
+
+	switch unit {
+	case 'H', 'h':
+		return time.Date(
+			now.Year(), now.Month(), now.Day(),
+			now.Hour()-amount, now.Minute(), now.Second(), now.Nanosecond(),
+			now.Location()), nil
+	case 'D', 'd':
+		return todayAtMidnight.AddDate(0, 0, -1*amount), nil
+	case 'W', 'w':
+		return todayAtMidnight.AddDate(0, 0, amount*-7), nil
+	case 'M', 'm':
+		return todayAtMidnight.AddDate(0, -1*amount, 0), nil
+	case 'Y', 'y':
+		return todayAtMidnight.AddDate(-1*amount, 0, 0), nil
+	default:
+		return todayAtMidnight, fmt.Errorf("invalid date format %s\nsee \"sweet stats --help\" for more details", arg)
+	}
+}
+
+// Writes a query that retrieves the entries specified by the start and end
+// date functions. Also handles the `since` variable, which is an alias for
+// start.
+func argsToQuery(cmd *cobra.Command, now time.Time) (string, error) {
+	filters := []string{}
+	name := cmd.Flag("name").Value.String()
+	lang := cmd.Flag("lang").Value.String()
+
+	if name != "" && lang != "" {
+		return "", fmt.Errorf("both name and lang provided (please pick one of them!)")
+	} else if lang != "" {
+		filters = append(filters, fmt.Sprintf("lang='%s'", lang))
+	} else if name != "" {
+		nameFilter := fmt.Sprintf("name like '%s'", name)
+		nameFilter = strings.Replace(nameFilter, "*", "%", -1)
+		filters = append(filters, nameFilter)
+	}
+
+	end := cmd.Flag("end").Value.String()
+	since := cmd.Flag("since").Value.String()
+	start := cmd.Flag("start").Value.String()
+
+	if end != "" && since == "" && start == "" {
+		return "", fmt.Errorf("must define start if end is provided")
+	}
+
+	if since != "" && start != "" {
+		fmt.Printf("both `--since` and `--start` variables provided (you only need one of them!)")
+	} else if since != "" && start == "" {
+		start = since
+	}
+
+	startTime, _ := parseDateFromArg(true, start, now)
+	filters = append(filters, fmt.Sprintf("start >= %d", startTime.UnixMilli()))
+	endTime, _ := parseDateFromArg(false, end, now)
+	filters = append(filters, fmt.Sprintf("end <= %d", endTime.UnixMilli()))
+	// TODO finish creating filters for language and name properties
+
+	if endTime.Before(startTime) {
+		return "", fmt.Errorf("end is before start")
+	}
+
+	filterText := strings.Join(filters, " and ")
+
+	query := fmt.Sprintf("select * from reps where %s order by start desc;", filterText)
+	return query, nil
+}
+
+// Converts a query to an array of Reps, which are the individual
+// rows that appear in stats database
+func queryToReps(query string) (reps []Rep, err error) {
```

Comment:
This might be too much abstraction, consider moving this code to the Cmd run function instead.